core/src/marker/markerManager.cpp

```diff
@@ -297,13 +298,36 @@ bool MarkerManager::buildStyling(Marker& marker) {
     if (!m_scene) { return false; }
 
     std::vector<StyleParam> params;
+
+    const auto& markerStyling = marker.styling();
+    const char* delimiter = ":";
```

Comment:
Since its the layers we are using for marker styling here, I think its redundant to add `layers`. So I have not modified this behavior, but included the use of `.` as a delimiter instead of `:`.